src/widget/wtrackmenu.cpp

```diff
@@ -2222,27 +2222,31 @@ void WTrackMenu::slotRemove() {
         return;
     }
     m_pTrackModel->removeTracks(getTrackIndices());
+    emit restorePreviousSelectionPos();
```

Comment:
I also considered to emit this signal from the trackmodels, but I didn't do it because a) there are no signals yet emitted from them (for a reason?), b) don't link GUI and track models? and c) looks like that would be even more `emit` calls than now.
So I think having these calls only in WTrackTableView and WTrackMenu is easily maintainable.